### ISSUE TITLE
Performance of Tensorflow distributed training  is not enough

### ISSUE THREAD

**Issue by ZhuFengdaaa** [tensorflow/models#101]
I have two machine, both of them have totally same environment: CentOS 7, 4 GTX TITAN X GPUs, 32 Intel i7 CPUs and 131GB Memory. The distributed training code is the newest master branch [here](https://github.com/tensorflow/models/tree/master/inception). The training speed is 

    INFO:tensorflow:Worker 6: 2016-05-16 21:07:22.101672: step 390, loss = 8.15(24.0 examples/sec; 1.334  sec/batch)
    INFO:tensorflow:Worker 5: 2016-05-16 21:07:22.101666: step 390, loss = 8.10(24.0 examples/sec; 1.335  sec/batch)
    INFO:tensorflow:Worker 4: 2016-05-16 21:07:22.101768: step 390, loss = 8.11(24.0 examples/sec; 1.333  sec/batch)
    INFO:tensorflow:Worker 7: 2016-05-16 21:07:22.102245: step 390, loss = 8.03(24.1 examples/sec; 1.330  sec/batch)

It is not satisfying because this speed is twice as slow as Multi-GPU training on one machine. I used `nvidia-smi -l 1` to watche the GPU usage, and find that GPUs are free in 3/4 running period. The `top` command print like this:

     %Cpu(s):  2.4 us, 35.2 sy,  0.0 ni, 62.3 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
    KiB Mem : 13174038+total, 21064972 free, 18241616 used, 92433792 buff/cache
    KiB Swap: 16777212 total, 16724332 free,    52880 used. 88948128 avail Mem 

      PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND     
      685 peter+  20   0  0.191t 2.336g 453624 S 878.1  1.9  83:03.03 python      
    15102 peter+  20   0 13.715g 5.522g  41820 S 293.7  4.4 348:39.74 python      
      682 peter+  20   0  0.192t 2.476g 453608 S  13.0  2.0  69:18.00 python      
      683 peter+  20   0  0.193t 4.093g 978228 S  11.6  3.3  84:40.56 python      
      684 peter+  20   0  0.191t 2.410g 453612 S  10.6  1.9  88:50.77 python 

I read file from local disk. 

I refered to this [issue](https://github.com/tensorflow/models/issues/47) and wondering if the bottle neck is CPU/IO. I tried to modify the `num_preprocess_threads` and the `num_readers` and find the best performance got `1.333  sec/batch`, as the speed posted above, is when I set both variable to 1. Modify the queue capacity is not helpful. I believe the `bath_inputs()` is executed on `CPU:0`, because in `image_processing.py` line 132:

     with tf.device('/cpu:0'):
        images, labels = batch_inputs(

Even though I don't know how to prove it. 

Then I don't know how to get a better performance.

**Comment by girving** [comment]
This is the wrong repository.  Please file TensorFlow issues at https://github.com/tensorflow/tensorflow/issues.

**Comment by ZhuFengdaaa** [comment]
Thanks, I have migrate this issue to [tensorflow repository](https://github.com/tensorflow/tensorflow/issues/2397). But I'm still wondering if we can modify the `inception_distributed_train.py` a bit and improve the training performance ?
